TACC/cms-site-resources
# GH-24: Updates for TACC/Core-CMS#148

# Overview

- Use `html-elements` stylesheet, not individual element stylesheets (to match https://github.com/TACC/Core-CMS/issues/148).
- Use `GH-25` instead of `HACK` to highlight undesired `#content-wrap`.

# Issues

- https://github.com/TACC/cms-site-resources/issues/24 (based on TACC/Core-CMS#148)

# Changes

1. Move elements styles from `generic/roots` (wrong folder) to `elements/html-elements` for:
    - `example-cms`
    - `neuronex-cms`
    - `protx-cms`
1. Move elements styles from separate files to `elements/html-elements` for `texascale-org`.
1. Add missing comment to one stylesheet in `texascale-org`.

# Testing

1. Update CMS to run `example-cms`.
    <details><summary>Steps</summary>

    1. Update `_CMS_TEMPLATES` in `secrets.py` to equal [that of `example-cms`'s](https://github.com/TACC/cms-site-resources/blob/task/GH-24-updates-for-core-cms-GH-148/example-cms/secrets.py#L19-L30).
    2. Update `CUSTOM_ASSET_DIR` in `.env` to equal `example-cms`.
    3. Run `npm run build` (in `cms` docker container if you don't install Node locally)
    4. Load any page.
    5. Set template to "Fullwidth".

    </details>
1. Confirm that previous element styles are _still_ applied.
    <details><summary>Styles</summary>

    ```
    body {
      background-color: yellow;
    }
    ```

    </details>
1. Update `CUSTOM_ASSET_DIR` in `.env` to equal `texascale-org`.
    <details><summary>Steps</summary>

    1. Update `_CMS_TEMPLATES` in `secrets.py` to equal [that of `texascale-org`'s](https://github.com/TACC/cms-site-resources/blob/task/GH-24-updates-for-core-cms-GH-148/texascale-org/secrets.py#L10-L18).
    2. Update `CUSTOM_ASSET_DIR` in `.env` to equal `texascale-org`.
    3. Run `npm run build` (in `cms` docker container if you don't install Node locally)
    4. Load any page.
    5. Set template to "Fullwidth".

    </details>
1. Confirm that previous element styles are _still_ applied.
    <details><summary>Styles</summary>

    ```
    […]
    #content-wrap {
      padding-bottom: var(--global-space-beneath-main-content);
    }
    ```

    </details>
